src/pyrokinetics/diagnostics/saturation_rules.py

```diff
@@ -0,0 +1,247 @@
+import numpy as np
+import xarray as xr
+from scipy.integrate import cumulative_trapezoid
+
+from ..pyroscan import PyroScan
+
+
+class SaturationRules:
+    r"""
+    Contains all the different saturation rules that can be applied
+
+    Need a PyroScan object to apply the rule to
+    """
+
+    def __init__(self, pyro_scan: PyroScan):
+
+        self.pyro_scan = pyro_scan
+
+    def mg_saturation(
+        self,
+        Q0: float = 25.0,
+        alpha: float = 2.5,
+        gamma_exb: float = 0.0,
+        output_convention: str = "pyrokinetics",
+        gamma_tolerance: float = 0.001,
+        equal_arc_theta: bool = True,
+        ky_dim: str = "ky",
+        theta0_dim: str = "theta0",
+    ):
```

Comment:
I would suggest adding the following header to this function
```
"""
Please see doi:10.1017/S0022377824001107 for details on this quasi-linear model.
"""
```
or something similar.